### PR TITLE
feat(iOS): scroll edge appearance

### PR DESCRIPTION
**Pull request by dylancom**
In many iOS apps such as the health app, the tabbar becomes transparent (without border).
In this PR I am looking into how we can add this.
It currently seems to bug at first render (as the comment also says: _This causes issues with lazy loading making the TabView background blink._), but if you scroll once the effect comes in.

| **Transparent without content overlapping** | **Content overlap** |
|---------------------------------------------|---------------------|
| Simulator Screenshot - iPhone 16 Pro - 2024-10-09 at 08 24 45 | Simulator Screenshot - iPhone 16 Pro - 2024-10-09 at 08 25 37 |

### CODE REVIEW

**Comment by okwasniewski** [comment]
Hey! Thanks for opening the PR.

I definitely want to support this but with lazy loading the tab bar was flashing before the content rendered.

I think we could make this into a prop, something like: `showTabBarBackground` (or something similar) and by default manipulate this setting on the JS side to prevent flashing if user is using lazy loading.

WDYT?

**Comment by dylancom** [comment]
@okwasniewski I am a bit unsure how to call this. Should we have a prop mimic the native side? `scrollEdgeAppearance="default | transparent"`.

_scrollEdgeAppearance: Describes the appearance attributes for the tabBar to use when an observable scroll view is scrolled to the bottom. If not set, standardAppearance will be used instead._

**Comment by okwasniewski** [comment]
@dylancom To me scrollEdgeAppearance="default | transparent" makes sense. This will be iOS only so using native name might be easier

File: src/TabView.tsx [hunk 1]
```diff
@@ -36,6 +36,10 @@ interface Props<Route extends BaseRoute> {
    * Whether to disable page animations between tabs. (iOS only)
    */
   disablePageAnimations?: boolean;
+  /**
+   * Describes the appearance attributes for the tabBar to use when an observable scroll view is scrolled to the bottom. (iOS only)
+   */
```
Comment (okwasniewski) [inline src/TabView.tsx:41]: Can you also add this to the docs? 

https://github.com/okwasniewski/react-native-bottom-tabs/blob/main/docs/docs/docs/guides/usage-with-react-navigation.md

File: ios/TabViewProvider.swift [hunk 2]
```diff
@@ -59,6 +59,12 @@ struct TabData: Codable {
     }
   }
   
+@objc var scrollEdgeAppearance: NSString? {
+    didSet {
+      props.scrollEdgeAppearance = scrollEdgeAppearance as? String
+    }
+  }
```
Comment (okwasniewski) [inline ios/TabViewProvider.swift:66]: Something is off with formatting here

File: ios/TabViewImpl.swift [hunk 3]
```diff
@@ -73,8 +74,16 @@ struct TabViewImpl: View {
...
+        let scrollEdgeAppearance = UITabBarAppearance()
+        switch props.scrollEdgeAppearance {
+        case "opaque":
+          scrollEdgeAppearance.configureWithOpaqueBackground()
+        case "transparent":
+          scrollEdgeAppearance.configureWithTransparentBackground()
+        default:
+          scrollEdgeAppearance.configureWithDefaultBackground()
+        }
+        UITabBar.appearance().scrollEdgeAppearance = scrollEdgeAppearance
```
Comment (okwasniewski) [inline ios/TabViewImpl.swift:86]: Can you move this to a helper function?

File: ios/RCTTabViewViewManager.mm [hunk 4]
```diff
@@ -31,5 +31,6 @@ - (UIView *)view
 RCT_EXPORT_VIEW_PROPERTY(labeled, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(ignoresTopSafeArea, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(disablePageAnimations, BOOL)
+RCT_EXPORT_VIEW_PROPERTY(scrollEdgeAppearance, NSString)
```
Comment (okwasniewski) [inline ios/RCTTabViewViewManager.mm:34]: You also need to add it to JS Spec here: https://github.com/okwasniewski/react-native-bottom-tabs/blob/95782ac53c09dd5d42b2ece0aae877310c762857/src/TabViewNativeComponent.ts#L18

**Review by okwasniewski — COMMENTED** [review]
Thanks for working on this! Looks good, I left few comments

**Comment by dylancom** [comment]
Created a new clean PR, see: #54

**Pull request closed without merging**